### PR TITLE
Updates to support Durable Client feature

### PR DESCRIPTION
**Pull request by bachuv**
These changes are for the new Durable Client feature (#1125). With the new feature, there isn't a way to create the HTTP management payloads from the lack of a webhook URL so exceptions are thrown when a user tries to call `CreateCheckStatusResponse` or `CreateHttpManagementPayload`.

### CODE REVIEW

File: src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs [hunk 1]
```diff
@@ -651,6 +651,11 @@ private static async Task<OrchestrationState> GetOrchestrationInstanceStateAsync
             string taskHubName,
             string connectionName)
         {
+            if (this.httpApiHandler == null)
+            {
+                throw new InvalidOperationException("CreateHttpManagementPayload not supported.");
```
Comment (ConnorMcMahon) [inline src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs:656]: Consider replacing the message with something like:

"IDurableClient.CreateHttpManagementPayload is not supported for IDurableClient instances created outside of an Azure Functions application."

File: src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs [hunk 2]
```diff
@@ -807,6 +812,11 @@ private static async Task<DurableOrchestrationStatus> GetDurableOrchestrationSta
             DurableClientAttribute attribute,
             bool returnInternalServerErrorOnFailure = false)
         {
+            if (this.httpApiHandler == null)
+            {
+                throw new InvalidOperationException("CreateCheckStatusResponse not supported.");
```
Comment (ConnorMcMahon) [inline src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs:817]: similar here.

File: src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClientFactory.cs [hunk 3]
```diff
@@ -76,19 +76,12 @@ public IDurableClient CreateClient(DurableClientOptions durableClientOptions)
...
-                    return new HttpApiHandler(null, null, this.durableTaskOptions, this.logger);
-                });
-
             DurableClient client = this.cachedClients.GetOrAdd(
                 attribute,
                 attr =>
                 {
                     DurabilityProvider innerClient = this.durabilityProviderFactory.GetDurabilityProvider(attribute);
-                    return new DurableClient(innerClient, httpApiHandler, attribute, this.MessageDataConverter, this.TraceHelper, this.durableTaskOptions);
+                    return new DurableClient(innerClient, null, attribute, this.MessageDataConverter, this.TraceHelper, this.durableTaskOptions);
```
Comment (ConnorMcMahon) [inline src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClientFactory.cs:84]: I believe this factory is also the implementation of  `IDurableClientFactory` for DurableTaskExtension, so I don't think we can safely set `httpApiHandler` to null here always. 

Is there a way to safely distinguish when we are running in a `DurableTaskExtension` and we can safey provide a `HttpApiHandler` instance, and when we can just specify `null`?

**Review by ConnorMcMahon — CHANGES_REQUESTED** [review]
I think the current iteration will break clients used by Durable Functions apps.

File: src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClientFactory.cs (hunk 3 above)
Comment (bachuv) [reply src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClientFactory.cs:84]: Can you point to where `DurableTaskExtension` is using `IDurableClientFactory`?

`DurableClientAttribute.IsExternalClient` could be a way to check if httpApiHandler should be set to null or not.

Comment (ConnorMcMahon) [reply src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClientFactory.cs:84]: Woops, I am mistaken! I assumed it got injected via dependency injection in DurableTaskExtension, but it looks like that is not the case? 

I am a bit confused of the need for `DurableClientAttrbiute.ExternalClient` then, but if I remember correctly that may be a remnant from some work @amdeel did a few releases back, not this new DurableClientFactory work...

File: test/Common/DurableClientBaseTests.cs [hunk 4]
```diff
@@ -142,6 +144,91 @@ public async Task TerminateAsync_NonRunningOrchestrator_ThrowsException()
         }
 
 #if !FUNCTIONS_V1
+        [Fact]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public async Task DurableClient_DI_StartNewAsync_ReturnsInstanceId()
```
Comment (ConnorMcMahon) [inline test/Common/DurableClientBaseTests.cs:149]: nit: I don't think the names of these tests are quite accurate, as it implies we are testing the full `IDurableClientFactory` code path, which we aren't quite doing. I would maybe mark them as "DurableClient_ExternalApp_*` instead. It's not perfect, but a bit more accurate.

**Review by ConnorMcMahon — APPROVED** [review]
I have a small nit regarding test names, but overall I think this is ready, at least for initial release!

**Pull request merged**